packages/next/client/experimental-script.tsx

```diff
@@ -0,0 +1,169 @@
+import React, { useEffect, useContext } from 'react'
+import { ScriptHTMLAttributes } from 'react'
+import { HeadManagerContext } from '../next-server/lib/head-manager-context'
+
+const ScriptCache = new Map()
+const LoadCache = new Set()
+
+const DOMAttributeNames: Record<string, string> = {
+  acceptCharset: 'accept-charset',
+  className: 'class',
+  htmlFor: 'for',
+  httpEquiv: 'http-equiv',
+}
+
+interface Props extends ScriptHTMLAttributes<HTMLScriptElement> {
+  strategy?: 'defer' | 'lazy' | 'dangerousRenderBlocking' | 'eager'
+  key?: string
+  onLoad?: () => void
+  onError?: () => void
+  children?: any
+  preload?: boolean
+}
+
+const onErrorHandler: Function = (src: string, onError: Function) => () => {
+  ScriptCache.delete(src)
+
+  if (onError) {
+    onError()
+  }
+}
```

Comment:
Why does a script error remove it from the script cache? This doesn't mimic normal browser behavior AFAIK and would allow users to keep trying to execute the script. Every single render would re-inject it, right?

Reply:
Thats a good point. Yes, each render would trigger the script. It would make it harder to retry loading the script if we don't undo the cache entry.